File: Plugins/Aspose_OCR_Java_for_Ruby/lib/asposeocrjava/OCR/multipagetiff.rb
module Asposeocrjava
  module MultipageTiff
    def initialize()    
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Initialize an instance of OcrEngine
        ocr_engine = Rjb::import('com.aspose.ocr.OcrEngine').new

        # Set the Image property by loading the image from file path location
        ocr_engine.setImage(Rjb::import('com.aspose.ocr.ImageStream').fromFile(data_dir + 'multipage.tiff'))

        # Set OcrEngine.ProcessAllPages to true in order to process all pages of TIFF in single run
        ocr_engine.setProcessAllPages(true)

        # Process the image
        if ocr_engine.process()
            # Retrieve the list of Pages
            pages = ocr_engine.getPages()
            pages.each do |page|
                puts "Text: " + page.getPageText().to_string
            end
        end
    end
  end
end

File: Plugins/Aspose_OCR_Java_for_PHP/src/aspose/ocr/WorkingWithOCR/MultipageTiff.php
<?php
namespace Aspose\OCR\WorkingWithOCR;

use com\aspose\ocr\OcrEngine as OcrEngine;
use com\aspose\ocr\ImageStream as ImageStream;
class MultipageTiff{
    public static function run($dataDir=null){

        # Initialize an instance of OcrEngine
        $ocr_engine = new OcrEngine();

        # Set the Image property by loading the image from file path location
        $imageStream=new ImageStream();
        $ocr_engine->setImage($imageStream->fromFile($dataDir . 'multipage.tiff'));

        # Set OcrEngine.ProcessAllPages to true in order to process all pages of TIFF in single run
        $ocr_engine->setProcessAllPages(true);

        # Process the image
        if ($ocr_engine->process()) {
            # Retrieve the list of Pages
        $pages = $ocr_engine->getPages();
            foreach($pages as $page) {
            print "Text: " . (string)$page->getPageText().PHP_EOL;
            }
        }
    }
}
?>